unstructured/documents/email_elements.py

```diff
@@ -42,6 +42,21 @@ def __init__(
         if isinstance(datestamp, datetime):
             self.datestamp: datetime = datestamp
 
+    def id_to_hash(self, index_in_sequence: int) -> str:
```

Comment:
I discovered that the `Name` and `EmailElement` are almost identical copies of `Text` and `Element`. Honestly, I don't like that design, it requires a refactor - should create a separate issue/PR for thath